lib/src/core/engine.dart

```diff
@@ -1037,7 +1037,18 @@ class Engine extends Disposable with EventsEmittable<EngineEvent> {
             return;
           }
           await signalClient.cleanUp();
-          await cleanUp();
+          if ([
```

Comment:
disconnect reason actually won't matter. the fields that we should use are:
* [can_reconnect](https://github.com/livekit/protocol/blob/main/protobufs/livekit_rtc.proto#L263)
* [action](https://github.com/livekit/protocol/blob/main/protobufs/livekit_rtc.proto#L265)

reason is just a informational string. what we care about is what the server-indicated action

Reply:
Solved, tested in e2e-flutter.